lib/puppet/defaults.rb

```diff
@@ -32,6 +32,41 @@ def self.valid_file_checksum_types
       %w[sha256 sha256lite sha384 sha512 sha224 sha1 sha1lite md5 md5lite mtime ctime]
   end
 
+  def self.log_ca_migration_warning
+    urge_to_migrate = <<-UTM
+The cadir is currently in your puppet/ssl directory. This location will be
+deprecated and removed in a future version of puppet. Please run the the
+puppetserver ca tool to migrate it out to the puppetserver directory. Use
+`puppetserver ca migrate --help` for more info.
+UTM
+    Puppet.warn_once('deprecations',
+                     'CA migration message',
+                     urge_to_migrate,
+                     :default,
+                     :default)
+  end
+
+  def self.default_cadir
+    old_ca_dir = "#{Puppet[:ssldir]}/ca"
+    new_ca_dir = "/etc/puppetlabs/puppetserver/ca"
```

Comment:
Previously `Puppet[:cadir]` resolved to `~/.puppetlabs/etc/puppet/ssl/ca` when running as non-root. Should a similar behavior occur when running as non-root, something like `~/.puppetlabs/etc/puppetserver/ca`?

Reply:
I think we want to stick with the hard-coded path because I don't think we really have a good use case for putting the cadir in some user path, it seems like that would happen in error more than someone actually wanting it in that location.

Reply:
I don't think there's a great user facing reason, but I do think this could break our dev workflow

Reply:
Yeah, but it should be easily fixable by specifying the cadir?